File: src/main/java/problems/easy/ClimbingStairs/ClimbingStairs.java
package problems.easy.ClimbingStairs;

@SuppressWarnings("PMD")
class Solution {
    public int climbStairs(int n) {
        if (1 == n)
            return 1;
        if (2 == n)
            return 2;
        var values = new int[n];
        values[0] = 1;
        values[1] = 2;
        if (2 < n) {
            for (var i = 2; i < n; i++)
                values[i] = values[i - 1] + values[i - 2];
        }
        return values[n - 1];
    }
}

File: src/main/java/problems/easy/ClimbingStairs/ClimbingStairs.kt
package problems.easy.ClimbingStairs

class SolutionKt {
    fun climbStairs(n: Int): Int {
        if (1 == n)
            return 1
        if (2 == n)
            return 2
        var values = IntArray(n)
        values[0] = 1
        values[1] = 2
        for (i in 2..n - 1)
            values[i] = values[i - 1] + values[i - 2]
        return values[n - 1]
    }
}
